huaweicloud/services/ddm/resource_huaweicloud_ddm_schema.go

```diff
@@ -0,0 +1,489 @@
+			"name":   utils.PathSearch("name", v, nil),
+			"status": utils.PathSearch("status", v, nil),
+		})
+	}
+	return rst
+}
+
+func resourceDdmSchemaDelete(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	cfg := meta.(*config.Config)
+	region := cfg.GetRegion(d)
+
+	// deleteSchema: Delete DDM schema
+	var (
+		deleteSchemaHttpUrl = "v1/{project_id}/instances/{instance_id}/databases/{ddm_dbname}"
+		deleteSchemaProduct = "ddm"
+	)
+	deleteSchemaClient, err := cfg.NewServiceClient(deleteSchemaProduct, region)
+	if err != nil {
+		return diag.Errorf("error creating DDM client: %s", err)
+	}
+
+	parts := strings.SplitN(d.Id(), "/", 2)
+	if len(parts) != 2 {
+		return diag.Errorf("invalid id format, must be <instance_id>/<db_name>")
+	}
+	instanceID := parts[0]
+	schemaName := parts[1]
+	deleteSchemaPath := deleteSchemaClient.Endpoint + deleteSchemaHttpUrl
+	deleteSchemaPath = strings.ReplaceAll(deleteSchemaPath, "{project_id}", deleteSchemaClient.ProjectID)
+	deleteSchemaPath = strings.ReplaceAll(deleteSchemaPath, "{instance_id}", fmt.Sprintf("%v", instanceID))
+	deleteSchemaPath = strings.ReplaceAll(deleteSchemaPath, "{ddm_dbname}", schemaName)
+
+	deleteSchemaQueryParams := buildDeleteSchemaQueryParams(d)
+	deleteSchemaPath += deleteSchemaQueryParams
+
+	deleteSchemaOpt := golangsdk.RequestOpts{
+		KeepResponseBody: true,
+		OkCodes: []int{
+			200,
+		},
+		MoreHeaders: map[string]string{
+			"Content-Type": "application/json",
+		},
+	}
+	for {
+		_, err := deleteSchemaClient.Request("DELETE", deleteSchemaPath, &deleteSchemaOpt)
+		if err == nil {
+			break
+		}
+		if errCode, ok := err.(golangsdk.ErrUnexpectedResponseCode); ok && errCode.Actual == 409 {
+			var apiError ddmError
+			err := json.Unmarshal(errCode.Body, &apiError)
+			if err != nil {
+				return diag.FromErr(err)
+			}
+			if apiError.ErrorCode == "DBS.200019" {
+				err = waitForInstanceRunning(ctx, d, meta, instanceID, []string{"DROP_DATABASE", "CREATE_DATABASE", "BACKUP"})
+				if err != nil {
+					return diag.FromErr(err)
+				}
+				continue
+			}
+		}
+		return diag.Errorf("error deleting DdmSchema: %s", err)
+	}
+
+	err = waitForInstanceRunning(ctx, d, meta, instanceID, []string{"DROP_DATABASE", "BACKUP"})
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	return nil
+}
+
+func buildDeleteSchemaQueryParams(d *schema.ResourceData) string {
+	res := ""
+	if v, ok := d.GetOk("delete_rds_data"); ok {
+		res = fmt.Sprintf("%s&delete_rds_data=%v", res, v)
+	}
+	if res != "" {
+		res = "?" + res[1:]
+	}
+	return res
+}
+
+func waitForInstanceRunning(ctx context.Context, d *schema.ResourceData, meta interface{}, instanceID string, pending []string) error {
+	cfg := meta.(*config.Config)
+	region := cfg.GetRegion(d)
+
+	stateConf := &resource.StateChangeConf{
+		Pending:      pending,
+		Target:       []string{"RUNNING"},
+		Refresh:      ddmInstanceStatusRefreshFunc(instanceID, region, cfg),
+		Timeout:      d.Timeout(schema.TimeoutDelete),
```

Comment:
the Timeout should not be fixed to TimeoutDelete

Reply:
done